### PR TITLE
Add OpenAPI validation in CRD schema

### PR DESCRIPTION
**Pull request by jessesuen**
Resolves https://github.com/argoproj/argo-cd/issues/1245

This change adds OpenAPI schema validation to the Application and AppProject CRD specs. This makes it possoble for the K8s API server to reject malformed CR objects from entering the system and crashing the controller. Needed to update our codegen dependencies as a result of this change.

### CODE REVIEW

File: hack/generate-proto.sh [hunk 1]
```diff
@@ -33,7 +33,6 @@ APIMACHINERY_PKGS=(
     k8s.io/api/core/v1
 )
 go-to-protobuf \
-    --logtostderr \
```
Comment (alexec) [inline hack/generate-proto.sh]: accident?

**Review by alexec — COMMENTED** [review]
I'm not sure what this PR is for. Any chance of some description please?

File: hack/generate-proto.sh (hunk 1 above)
Comment (jessesuen) [reply hack/generate-proto.sh]: The newer codegen utilities removed this flag as a result of the switch from glog to klog, so I needed to remove it.

**Review by alexec — APPROVED** [review]

**Review by alexmt — APPROVED** [review]
LGTM

**Pull request merged**